R/mlr3learners_table.R

```diff
@@ -0,0 +1,130 @@
+# Ideally this table would be created automatically and all required packages would be installed.
+#  and loaded. Required packages are mlr3, mlr3learners, mlr3proba, and all packages in
+#  mlr3learners org, also when ready other packages in mlr3verse that have learners implemented in
+#  them.
+
+# library(mlr3)
+# library(mlr3learners)
+# library(mlr3proba)
+# library(data.table)
+extra_learners = rownames(
+  available.packages(repos = "https://mlr3learners.github.io/mlr3learners.drat")
+  )
```

Comment:
Suggestion: Wrap in `tryCatch()` because it should not block if no internet is available.

Reply:
If this is automated and created during builds then surely there is always internet?

Reply:
Ah I did not see that this is not part of the function.

We could possibly store it in the package though the content then depends on the version which users have installed.

I guess querying an online resource (could also be the mlr3 GH repo) while requiring internet access would be better? What about both: Trying to query the online table and fall back to the local one included in the package (with a warning message that this one might not include all learners).

Reply:
> I guess querying an online resource (could also be the mlr3 GH repo) while requiring internet access would be better? 

What's/where's the mlr3 GH repo? 

> all back to the local one included in the package (with a warning message that this one might not include all learners).

I think this would slightly defeat the point because say a user wants to install xgboost but does not know that it lives in mlr3learners and have therefore only installed mlr3. Then the table will not show xgboost nor will it be able to install it if called.

Unless you just mean *only* fall back to local when internet is not available? I guess that would make sense and be more intuitive than just erroring. Assuming the local one is identical to the code above except with no call to `install.packages`/`available.packages`?

Reply:
> Unless you just mean only fall back to local when internet is not available?

I meant this, yes.

> What's/where's the mlr3 GH repo?

The mlr3 GitHub repo.

Outlining the process again:

1. `lrns()` queries an online resource which could live in the mlr3data repo (to not bloat mlr3 with too much CI updates) GitHub repo. This resource gets generated once a day.
2. If no internet access is available, a static version included in the mlr3 package installation is used. This option included a warning that a static version was used.
3. In every mlr3 release, mlr3 ships with the most recent version of this static learner table.

Reply:
Sorry completely misread and thought you were saying there's a separate repo just for certain GHaction automations. 
I completely agree with the process outlined above.  But would this not result in `mlr3data` becoming an import not suggest as the `lrn/lrns` functions would depend on this?

Reply:
- The latest version of the learner table (which would be updated by the CI every day) would live in mlr3data and would be queried via the web
- The static version would live in mlr3 and ship with the package

Reply:
That sounds good and just so I understand would the static version just have to be manually typed out and updated. Do you want me to try and set-up the build for mlr3data so we can close the first part (i.e. the online table)?

Reply:
First I would like to ask @mllg if he agrees that mlr3data would be a good place to store a .csv file containing this information?

> That sounds good and just so I understand would the static version just have to be manually typed out

Not sure I understand what you mean by this.
1. Read the .csv from mlr3data into an .rda file into mlr3
2. Ship this .rda file (static snapshot) with an mlr3 release and use it as the static fallback

Reply:
Got it, sorry I have no experience with external data structures in R so just trying to visualise it, but understand it now :)